### PR TITLE
Add "tflint" Hook

### PR DESCRIPTION
**Pull request by Tensho**

### CODE REVIEW

File: hooks/tflint.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+set -e
+
+# OSX GUI apps do not pick up environment variables the same way as Terminal apps and there are no easy solutions,
+# especially as Apple changes the GUI app behavior every release (see https://stackoverflow.com/q/135688/483528). As a
+# workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
+export PATH=$PATH:/usr/local/bin
+
+for file in "$@"; do
+  tflint `dirname $file`
```
Comment (yorinasub17) [inline hooks/tflint.sh]: Same issue as the terraform validate PR

**Comment by Tensho** [comment]
@yorinasub17 Sorry for the inconvenience, you are totally right. I did this way because I had to refer to the overall changes in my project somehow. But it's kind of selfish, there is no excuse for me. I'll split them up carefully.

File: hooks/tflint.sh (hunk 1 above)
Comment (Tensho) [reply hooks/tflint.sh]: `tflint` can work with files and directories. IMHO, if the tool may work with files, we should leverage on the `pre-commit`'s parallelization default.

Comment (yorinasub17) [reply hooks/tflint.sh]: Sounds good. The only reason why I highlighted that was because you were using `dirname`.

**Comment by yorinasub17** [comment]
> But it's kind of selfish, there is no excuse for me. I'll split them up carefully.

No worries! That was just a request.

> I did this way because I had to refer to the overall changes in my project somehow.

The workflow I use is to commit each change set individually (using `git add -p`), then create branches for each change and use `git cherry-pick`. That way, I have a branch with everything, but also the split versions. For a small number of change sets, it works well. It can be tedious with lots of changes though.

**Comment by yorinasub17** [comment]
Another approach is to work backwards:

- Have a branch with all the commits
- Create new branches for each change set to turn into PR from the combined branch
- Use `git rebase -i` to drop the irrelevant commits from each subbranch.

**Comment by Tensho** [comment]
@yorinasub17 Please let me know if I can do anything else.

**Comment by yorinasub17** [comment]
You'll need to resolve the conflict now that I merged the `terraform-validate` PR.

**Comment by Tensho** [comment]
@yorinasub17 Resolved

**Comment by Tensho** [comment]
@yorinasub17 Please let me know if I can do anything here 🙇

**Comment by Tensho** [comment]
@yorinasub17 That's OK ⛄️ ❄️ 🏂  Thank you 🙇  Happy New Year!

**Pull request merged**